From 377b27bf8e3f195fd34d20345dc8a28ddcc5193d Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:07:53 -0400
Subject: [usb] xhci: fix wrong port number reported when setting USB2.0 hardware LPM

Message-id: <1410887463-20674-402-git-send-email-dzickus@redhat.com>
Patchwork-id: 95591
O-Subject: [RHEL7 PATCH 401/591] xhci: fix wrong port number reported when setting USB2.0 hardware LPM.
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 654a55d34f880bb56b3cebab53771a09a404a1f8
Author: Lin Wang <bupt.wanglin@gmail.com>
Date:   Thu May 8 19:25:54 2014 +0300

    xhci: fix wrong port number reported when setting USB2.0 hardware LPM.

    This patch fix wrong port number reported when trying to enable/disable
    USB2.0 hardware LPM.

    Signed-off-by: Lin Wang <lin.x.wang@intel.com>
    Signed-off-by: Mathias Nyman <mathias.nyman@linux.intel.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/xhci.c b/drivers/usb/host/xhci.c
index 7bb3be8..49709c8 100644
--- a/drivers/usb/host/xhci.c
+++ b/drivers/usb/host/xhci.c
@@ -4092,7 +4092,7 @@ int xhci_set_usb2_hardware_lpm(struct usb_hcd *hcd,
  field = le32_to_cpu(udev->bos->ext_cap->bmAttributes);
 
  xhci_dbg(xhci, "%s port %d USB2 hardware LPM\n",
-   enable ? "enable" : "disable", port_num);
+   enable ? "enable" : "disable", port_num + 1);
 
  if (enable) {
   /* Host supports BESL timeout instead of HIRD */
-- 
1.7.1